### PR TITLE
Transition to profile through link, POST request, validation on type of user

### PR DESCRIPTION
**Pull request by olenakachmar**
renamed create-user-page-component to create-update-user-page-compone…  …
…nt, add possibility of going on pages regarding links on side-bar-component, make checking of type of user to show or hide some fields, POST request (unfinished)

### CODE REVIEW

File: client/src/app/page/create-update-user-page/create-update-user-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export class CreateUpdateUserPageComponent implements OnInit {
+  newUser: User;
+
+  constructor(readonly userService: UserService) {
+  }
+
+  extractUser(user): any {
+    this.newUser = user;
+    this.userService.addUser(this.extractUser(this.newUser))
+      .subscribe((data: User) => {
```
Comment (idyac) [inline client/src/app/page/create-update-user-page/create-update-user-page.component.ts]: ?

File: client/src/app/page/user-bar-info/user-bar-info.component.html [hunk 2]
```diff
@@ -5,16 +5,21 @@
...
+                         [info]="userinfo.manager.firstName + ' ' + userinfo.manager.lastName"
+                         [label]="'My manager'">
+    </app-link-to-profile>
     <app-link-to-profile class="additional-info" [info]="userinfo.department.name"
-                         [label]="'My department'"></app-link-to-profile>
-    <app-link-to-profile class="additional-info" [info]="userinfo.myTeamlead"
-                         [label]="'My Team Lead'"></app-link-to-profile>
+                         [label]="'My department'">
+    </app-link-to-profile>
+    <app-link-to-profile *ngIf="show" class="additional-info" [idLinks]="userinfo.manager.teamlead"
```
Comment (idyac) [inline client/src/app/page/user-bar-info/user-bar-info.component.html]: use just userType !== 'hr' and remove it from js

File: client/src/app/page/create-update-user-page/create-update-user-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  extractUser(user): any {
+    this.newUser = user;
+    this.newUser.contacts = [new Contact('a', 'b')];
+    this.newUser.dates = ['16/04/2019'];
+    this.newUser.type = 'developer';
+    this.newUser.phone = '6448845';
+    this.newUser.email = 'so@gmail.com';
+    this.userService.addUser(this.newUser)
+      .subscribe((data: any) => {
+      window.location.href = `/profile/my-profile/${data.newUser._id}`;
```
Comment (idyac) [inline client/src/app/page/create-update-user-page/create-update-user-page.component.ts]: indent

Comment (olenakachmar) [reply client/src/app/page/create-update-user-page/create-update-user-page.component.ts]: what is that?

**Pull request merged**